### PR TITLE
Create scala-based tracking controller to listen to signals and streams

### PR DESCRIPTION
**Pull request by deanrobertcook**
Still needs tracking for like button clicks and double tap.

Still not compiling, but please take a look
#### APK
[Download build #8213](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8213/artifact/build/artifact/wire-dev-PR456-8213.apk)
[Download build #8215](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8215/artifact/build/artifact/wire-dev-PR456-8215.apk)
[Download build #8219](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8219/artifact/build/artifact/wire-dev-PR456-8219.apk)

### CODE REVIEW

File: app/src/main/scala/com/waz/zclient/messages/MessagesListAdapter.scala [hunk 1]
```diff
@@ -1,4 +1,21 @@
 /**
+ * Wire
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/messages/MessagesListAdapter.scala]: Remove double header

File: app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala [hunk 2]
```diff
@@ -64,18 +59,13 @@ class MessageActionsController(implicit injector: Injector, ctx: Context, ec: Ev
 
   val onMessageAction = EventStream[(MessageAction, Message)]()
 
+  val onDeleteConfirmed = EventStream[(Message, Boolean)]()
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala]: Is there a way to somehow explain what the `Boolean` is for? Now the only way to find out is to look in the tracking controller. I guess we should at least have a comment for this here.

File: app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala [hunk 3]
```diff
@@ -163,7 +149,6 @@ class MessageActionsController(implicit injector: Injector, ctx: Context, ec: Ev
       .show()
 
   private def forwardMessage(message: Message) = {
-    trackingController.tagEvent(new ForwardedMessageEvent(message.getMessageType.name))
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala]: Did you miss tracking for forwarded message here?

File: app/src/main/scala/com/waz/zclient/messages/parts/YouTubePartView.scala [hunk 4]
```diff
@@ -63,6 +63,8 @@ class YouTubePartView(context: Context, attrs: AttributeSet, style: Int) extends
   private val content = Signal[MessageContent]()
   private val width = Signal[Int]()
 
+  val message = Signal[MessageData]()
```
Comment (hakonbo) [inline app/src/main/scala/com/waz/zclient/messages/parts/YouTubePartView.scala]: I added this in `ClickableViewPart` in another PR, so perhaps change here or there.

File: app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala (hunk 3 above)
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala]: Nope, it's now in the (scala) `TrackingController` - I listen to the `messageAction` signal, which is what used to call `forwardmessage()`

File: app/src/main/scala/com/waz/zclient/messages/parts/YouTubePartView.scala (hunk 4 above)
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/messages/parts/YouTubePartView.scala]: Yeah, I think we should merge your stuff first

File: app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala (hunk 2 above)
Comment (deanrobertcook) [inline app/src/main/scala/com/waz/zclient/messages/controllers/MessageActionsController.scala]: Unfortunately not, comment is the best bet...

**Comment by hakonbo** [comment]
build again

**Comment by shumengye** [comment]
@deanrobertcook looks good, just one small typo:
when liking or unliking a message from the options menu, attribute action="other" instead of being like or unlike

**Pull request merged**